src/lib.rs

```diff
@@ -146,7 +146,15 @@ where
 
     fn call(&mut self, mut req: Request<B>) -> Self::Future {
         if let Ok(claim) = self.jwt.claim() {
-            req.headers_mut().insert(AUTHORIZATION, claim);
+            let auth_bytes = format!("{} ", Bearer::SCHEME)
+                .as_bytes()
+                .iter()
+                .chain(claim.as_bytes())
+                .cloned()
+                .collect::<Vec<_>>();
```

Comment:
This seems like an overkill. Move it to `JwtSecret::claim` so that header value is constructed there